go/worker/compute/executor/committee/node.go

```diff
@@ -982,27 +1017,26 @@ func (n *Node) startProcessingBatchLocked(batch *unresolvedBatch) {
 	blk := n.commonNode.CurrentBlock
 	consensusBlk := n.commonNode.CurrentConsensusBlock
 	height := n.commonNode.CurrentBlockHeight
+	roundResults := lastRoundResults
 	go func() {
 		defer close(done)
 
-		// Fetch message results emitted during the last normal round.
-		state, err := n.commonNode.Consensus.RootHash().GetRuntimeState(ctx, blk.Header.Namespace, height)
+		rtDesc, err := n.commonNode.Runtime.RegistryDescriptor(n.ctx)
```

Comment:
Hm, so in case of changes, roothash has a snapshot of the descriptor that is valid for the current round. Could also fetch this at the epoch transition and cache it?

Reply:
yeah actually i see that we also already watch descriptor updates in the worker loop (in case we need to update scheduler parameters), can probably get it there as well?

Reply:
Actually if we cached the runtime descriptor on epoch change we'd need to access the cached runtime only while having the CrossNode lock (or use a new lock for the descriptor - but this is what the common node runtime registry already does). Therefore i think it makes more sense to just query the common node runtime registry as we do now?  (note: in this specific case we already have the CrossNode lock before the goroutine, so it's not the best point, but there are other places in the executor where we query the runtime descriptor and we don't have the lock currently and would need it if we switched to caching the descriptor on epoch transitions).

Or is there any other benefit doing it the other way?

Reply:
So my main concern is that the registry descriptor may be different than what is actually in effect for the current epoch. This is because on each epoch transition the roothash service makes a snapshot of the runtime descriptor that is valid in that epoch. If the descriptor is then updated mid-epoch it may contain different information.

Reply:
Actually it would probably be better to update the `RegistryDescriptor` and related methods to return a snapshot at the start of an epoch?

Reply:
Ah i see. Yeah I was about to mention that in that case this is likely already problematic in some existing places where runtime descriptor is queried. I think updating the `RegistryDescriptor` is better. Need to just double-check if all existing places that use it will work as expected with the change.

Reply:
Yeah skimming over all places where we use either `RegistryDescriptor` or `WatchRegistryDescriptor` the correct (or intended) are he "currently active parameters" so querying the roothash is more appropriate. Will update.

Reply:
Actually there are few places that actually want the descriptor in registry (e.g. keymanager,storage,common committee when initializing - before the runtime is even initialized). So I added new methods to the runtime registry for fetching active descriptor (`ActiveDescriptor` `WatchActiveDescriptor`) and also kept the old ones.